packages/core/test/support.ts

```diff
@@ -278,12 +278,17 @@ export function getPoolMax(): number {
   return Config[getTestDialect()].pool?.max ?? 1;
 }
 
-type ExpectationKey = 'default' | Permutations<Dialect>;
+type ExpectationKey = 'default' | Permutations<Dialect, 4>;
 
 export type ExpectationRecord<V> = PartialRecord<ExpectationKey, V | Expectation<V> | Error>;
 
-type Permutations<T extends string, U extends string = T> =
-  T extends any ? (T | `${T} ${Permutations<Exclude<U, T>>}`) : never;
+type Decr = [never, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+type Permutations<T extends string, Depth extends number, U extends string = T> = Depth extends 0
+  ? never
+  : T extends any
+    ? T | `${T} ${Permutations<Exclude<U, T>, Decr[Depth]>}`
```

Comment:
```suggestion
type DecrementedDepth = [never, 0, 1, 2, 3];

type Permutations<T extends string, Depth extends number, U extends string = T> = Depth extends 0
  ? never
  : T extends any
    ? T | `${T} ${Permutations<Exclude<U, T>, DecrementedDepth[Depth]>}`
```
This is the complete suggestion, I think if we commit this (or something with a similar name) we should be good to merge

Reply:
Unit and typing tests pass so it seems to works fine